arches_component_lab/src/arches_component_lab/widgets/FileListWidget/components/FileListWidgetEditor.vue

```diff
@@ -0,0 +1,199 @@
+<script setup lang="ts">
+import { onMounted, useTemplateRef, ref, watch, computed } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Message from "primevue/message";
+import FileUpload, {
+    type FileUploadRemoveEvent,
+    type FileUploadSelectEvent,
+} from "primevue/fileupload";
+import Image from "primevue/image";
+import Button from "primevue/button";
+import { FormField } from "@primevue/forms";
+
+import type { FormFieldResolverOptions } from "@primevue/forms";
+import type { FileReference } from "@/arches_component_lab/widgets/types.ts";
+const { $gettext } = useGettext();
+const allowedFileTypes = ref();
+const props = defineProps<{
+    initialValue: FileReference[] | null | undefined;
+    graphSlug: string;
+    nodeAlias: string;
+    widgetData: {
+        config: {
+            acceptedFiles: string;
+            maxFilesize: number;
+            rerender: boolean;
+            label: string;
+        };
+    };
+    nodeData: {
+        config: {
+            imagesOnly: boolean;
+            maxFiles: number;
+            maxFileSize: number;
+        };
+    };
+}>();
+
+const formFieldRef = useTemplateRef("formFieldRef");
+const currentValues = ref<FileReference[]>();
+
+watch(
```

Comment:
I'd say watching initialValue is overkill here. Instead you could just set it in-line with the currentValues declaration. An initialValue change could trigger a rerender.

Reply:
I don't think this is necessary at all anymore; just a result of initial flailings.